flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/iterator/RocksStatesPerKeyGroupMergeIterator.java

```diff
@@ -106,7 +118,8 @@ public void next() {
                 detectNewKeyGroup(oldKey);
             }
         } else {
-            IOUtils.closeQuietly(rocksIterator);
+            IOUtils.closeQuietly(currentSubIterator);
+            closeableRegistry.unregisterCloseable(currentSubIterator);
```

Comment:
Usually the pattern is:

```
            if (closeableRegistry.unregisterCloseable(currentSubIterator)) {
                IOUtils.closeQuietly(currentSubIterator);
            }
```
BTW, do we even need to unregister it from the registry? I think it won't harm, worst case it will be closed twice (which was the case previously as well).

Reply:
Thanks, I will use this pattern! And yes, it's actually important to unregister, we have tests that verify that exactly the right amount of `close()` calls happen on the Rocks classes. I was surprised myself. 🎊